### PR TITLE
docs: rewrite HowToAddnewBoard.md

### PR DESCRIPTION
**Pull request by sunghan-chang**
Previous is not enough to know how to add a new board.
This commit describes details.

Signed-off-by: sunghan-chang <sh924.chang@samsung.com>

### CODE REVIEW

File: docs/HowToAddnewBoard.md [hunk 1]
```diff
@@ -1,93 +1,182 @@
...
-Example: *os/arch/arm/src/s5j*, *os/arch/arm/include/s5j*  
+This includes *CPU* and *chip* architecture inside.  
 
-Refer *[kconfig](#kconfig)* for adding Kconfig entries
+### CPU
 
-## Board
+```
+	.
+	`- os - arch - <cpu-name> .- include - <cpu-subname>
```
Comment (manoharahk) [inline docs/HowToAddnewBoard.md]: Hello @sunghan-chang ,

We also have cpu-family along with cpu-artchitecures.
In patch, os/arch/arm/src/armv7-r
cpu-family : arm 
cpu-architecture : armv7-r

**Comment by manoharahk** [comment]
Hello @sunghan-chang ,

We also have  cpu-family name along with cpu-architecture in the path.
In path, os/arch/arm/src/armv7-r

cpu-family : arm
cpu-architectures : armv7-r.

**Comment by seinfra** [comment]
Target : [ac4583654b30d39991a5044b2823ad7806ef4c89] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ac4583654b30d39991a5044b2823ad7806ef4c89] - Code Rule Check (C++) OK.

**Pull request merged**